src/rdb/rdb_raft.c

```diff
@@ -1146,7 +1140,7 @@ rdb_raft_log_offer_single(raft_server_t *raft, void *arg,
 				DP_DB(db), index, rc);
 			goto err_discard;
 		}
-	} else {
+	} else if (raft_entry_is_cfg_change(entry)) {
```

Comment:
Is it for motivated by the test bug? If we add this check, we'd better also add an "else { D_ASSERTF(0, "unknown entry type %d\n", entry->type) }".